src/psyclone/psyGen.py

```diff
@@ -6070,8 +6142,41 @@ def __str__(self):
             ret = ret[:-2] + "]"  # Deletes last ", " and adds "]"
         else:
             ret += "Scalar"
+        if self.is_constant:
+            ret += ", constant, value={0}".format(self.constant_value)
         return ret + ">"
 
+    def copy(self):
+        '''Create and return a copy of this object.
+
+        :returns: A symbol object with the same properties as this \
+        symbol object
+        :rtype: :py:class:`psyclone.psyGen.Symbol`
+
+        '''
+        return Symbol(self.name, self.datatype, shape=self.shape[:],
+                      scope=self.scope, constant_value=self.constant_value,
+                      is_input=self.is_input, is_output=self.is_output)
+
+    def set_properties(self, symbol_in):
+        '''Replace all properties in this object with the properties from
+        symbol_in, apart from the name which is immutable.
+
+        :param symbol_in: A symbol object
```

Comment:
Suggestion: "Symbol that the properties are copied from." Otherwise just the punctuation.

Reply:
Yes, that is better. Modified. I also added documentation for an exception that I'd missed.